### PR TITLE
Upgrade dependency cassandra 3 -> 4

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ public void close() {
             new ImageFromDockerfile(imageName,doNotDeleteImageAfterUsage)
                 .withDockerfileFromBuilder(builder ->
                     additionalSteps.applyStep(builder
-                        .from("cassandra:3.11.15")
+                        .from("cassandra:4.0")
```
Comment (chibenwa) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java:171]: 4.1.latest please

Comment (vttranlina) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java:171]: 4.1 is GA Version, 4.0 is Stable version,
is it ok with 4.1?

**Comment by vttranlina** [comment]
CI fail
```
Existing failures - 10
(?) – org.apache.james.jmap.cassandra.access.CassandraAccessTokenRepositoryTest
<1s
(?) – org.apache.james.jmap.cassandra.change.CassandraEmailChangeRepositoryTest
...
```
I tested it on my local, and it passes

**Comment by Arsnael** [comment]
I see in the CI logs => `Could not initialize class org.apache.james.backends.cassandra.DockerCassandraSingleton`

I think unrelated, I will restart your PR

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java (hunk 1 above)
Comment (chibenwa) [reply backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java:171]: Yes

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**